credential/credential.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package credential
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-leia"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/credential/concept"
+)
+
+// NewCredentialInstance creates a new credential instance with default config and empty concept registry
+func NewCredentialInstance() Credential {
+	return &credential{
+		config:   DefaultConfig(),
+		registry: concept.NewRegistry(),
+	}
+}
+
+type credential struct {
+	registry concept.Registry
+	config   Config
+	store    leia.Store
+}
+
+func (c *credential) Registry() concept.Registry {
+	return c.registry
+}
+
+func (c *credential) Reader() VCReader {
+	return c
+}
+
+func (c *credential) Configure(config core.ServerConfig) error {
+	var err error
+	fsPath := path.Join(config.Datadir, "credentials", "credentials.db")
+
+	if c.store, err = leia.NewStore(fsPath); err != nil {
+		return err
+	}
+
+	// load templates
+	if err = c.registry.LoadTemplates(); err != nil {
+		return err
+	}
+
+	// init indices
+	if err = c.initIndices(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *credential) initIndices() error {
+	for _, templates := range c.registry.ConceptTemplates() {
```

Comment:
some parts can be abstracted to a more `store` like interface. This is surely the case when supporting more than 1 DB type. But in this PR I felt that showing how everything works is more important given the number of lines it would take to add the abstraction.